crates/gossip/src/summary.rs

```diff
@@ -0,0 +1,194 @@
+                        ring_top_hashes,
+                    } => {
+                        summary.dht_summary.insert(
+                            match arc_set.as_arcs().first().ok_or_else(
+                                || K2Error::other("empty arc set"),
+                            )? {
+                                DhtArc::Arc(start, end) => {
+                                    format!("{}..{}", start, end)
+                                }
+                                DhtArc::Empty => {
+                                    return Err(K2Error::other("empty arc"))
+                                }
+                            },
+                            DhtSegmentState {
+                                disc_top_hash,
+                                disc_boundary,
+                                ring_top_hashes,
+                            },
+                        );
+                    }
+                    _ => {
+                        return Err(K2Error::other("unexpected snapshot type"))
+                    }
+                }
+            }
+        }
+
+        let agents = self.peer_store.get_all().await?;
+        for agent in agents {
+            if local_agents.iter().any(|l| l.agent() == &agent.agent) {
+                continue;
+            }
+
+            let Some(url) = agent.url.clone() else {
+                continue;
+            };
+
+            summary.peer_meta.insert(
+                url.clone(),
+                PeerMeta {
+                    last_gossip_timestamp: self
+                        .peer_meta_store
+                        .last_gossip_timestamp(url.clone())
+                        .await?,
+                    new_ops_bookmark: self
+                        .peer_meta_store
+                        .new_ops_bookmark(url.clone())
+                        .await?,
+                    peer_behavior_errors: self
+                        .peer_meta_store
+                        .peer_behavior_errors(url.clone())
+                        .await?,
+                    local_errors: self
+                        .peer_meta_store
+                        .local_errors(url.clone())
+                        .await?,
+                    peer_busy: self
+                        .peer_meta_store
+                        .peer_busy(url.clone())
+                        .await?,
+                    peer_terminated: self
+                        .peer_meta_store
+                        .peer_terminated(url.clone())
+                        .await?,
+                    completed_rounds: self
+                        .peer_meta_store
+                        .completed_rounds(url.clone())
+                        .await?,
+                    peer_timeouts: self
+                        .peer_meta_store
+                        .peer_timeouts(url.clone())
+                        .await?,
+                },
+            );
+        }
+
+        serde_json::to_value(summary)
```

Comment:
Can we return the typed response? As a precursor to typing the conductor response too.

Reply:
Saw your comment above and replied there.